lagom-cloud-object-storage-example/account-api/src/main/java/com/lightbend/lagom/account/api/Extract.java

```diff
@@ -13,20 +13,25 @@
   public final Integer extractNumber;
   public final double startBalance;
   public final double endBalance;
-  public final PCollection<TransactionEntry> transactionEntries;
+  public final String status;
+  public final PCollection<TransactionEntry> txEntries;
 
   @JsonCreator
   public Extract(String accountNumber,
                  Integer extractNumber,
                  double startBalance,
                  double endBalance,
-                 PCollection<TransactionEntry> transactionEntries) {
+                 String status,
```

Comment:
Yes, that's purely for simplicity. I didn't consider it very useful to have an enum on something that is purely meant to generate json.

Reply:
Using enums in JSON serialization types can be dangerous... makes evolution difficult because adding new values breaks deserialization on older versions.

Reply:
This is getting off track, but I've been wondering if we should add some kind of "`SafeEnum`" helper to Lagom for Java that is essentially `Either<String,E extends Enum<E>>` (or however you get that to compile 😛 ). I know there is no `Either` in Java, but I mean it would be a sum type that wraps an enum with a possibly unexpected value.